src/components/Nav/Nav.js

```diff
@@ -1,10 +1,113 @@
 import React from 'react';
+import { withRouter } from 'react-router-dom';
+import { Link } from 'react-router-dom';
+import Button from '../Button/Button';
 import './Nav.scss';
 
 class Nav extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      isOpen: false,
+      userLists: {},
+    };
+  }
+
+  hadleUserData() {
+    console.log('handleUserData실행');
+
+    fetch('/data/userData.json', {
+      method: 'GET', // GET method는 기본값이라서 생략이 가능합니다.
+    }) // 예시코드에서는 이해를 돕기 위해 명시적으로 기입해뒀습니다.
+      .then(res => res.json())
+      .then(data =>
+        this.setState({
+          userLists: data,
+        })
+      );
+  }
+
+  componentDidMount() {
+    this.hadleUserData();
+  }
+
+  openMenu = () => {
+    if (this.state.isOpen === false) {
+      this.setState({
+        isOpen: true,
+      });
+    } else {
+      this.setState({
+        isOpen: false,
+      });
+    }
+  };
+
+  hotItem = e => {
+    this.props.history.push('/articles');
+  };
+
+  goMypage = e => {
+    this.props.history.push('/mypage');
+  };
+
+  goLogin = e => {
+    this.props.history.push('/login');
+  };
+
   render() {
-    return <div />;
+    const { userLists } = this.state;
```

Comment:
```suggestion
    const { userName, userNickname} = this.state. userLists;
```
위와 같이 구조분해를 하면 좀 더 편하게 사용하실 수 있습니다!

Reply:
도현님 바로 적용하겠습니다! 감사드립니다!